pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        "",
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.NodeAffinity = newNonNilNodeAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil PodAffinity",
+			rack: func() scyllav1.RackSpec {
+				r := newBasicRack()
+				r.Placement = &scyllav1.PlacementSpec{
+					PodAffinity: newNonNilPodAffinity(),
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        "",
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.PodAffinity = newNonNilPodAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil PodAntiAffinity",
+			rack: func() scyllav1.RackSpec {
+				r := newBasicRack()
+				r.Placement = &scyllav1.PlacementSpec{
+					PodAntiAffinity: newNonNilPodAntiAffinity(),
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        "",
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.PodAntiAffinity = newNonNilPodAntiAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil ImagePullSecrets",
+			rack: newBasicRack(),
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := newBasicScyllaCluster()
+				sc.Spec.ImagePullSecrets = newNonNilImagePullSecrets()
+				return sc
+			}(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        "",
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Spec.ImagePullSecrets = newNonNilImagePullSecrets()
+				return sts
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil ForceRedeploymentReason",
+			rack: newBasicRack(),
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := newBasicScyllaCluster()
+				sc.Spec.ForceRedeploymentReason = "reason"
+				return sc
+			}(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        "",
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Annotations[naming.ForceRedeploymentReasonAnnotation] = "reason"
+				return sts
+			}(),
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := StatefulSetForRack(tc.rack, tc.scyllaCluster, tc.existingStatefulSet, tc.sidecarImage)
+
+			if !reflect.DeepEqual(err, tc.expectedError) {
+				t.Errorf("expected and actual errors differ: %s",
+					cmp.Diff(tc.expectedError, err, cmpopts.EquateErrors()))
```

Comment:
I guess with EquateErrors even suberrors would pass

Reply:
you're right, I removed cmopts.EquateErrors()